src/hotspot/share/oops/klass.hpp

```diff
@@ -511,18 +511,22 @@ class Klass : public Metadata {
     return (BasicType) btvalue;
   }
 
-  // Want a pattern to quickly diff against layout header in register
-  // find something less clever!
+  // VS warns (C4146) about unary minus of unsigned.
+  PRAGMA_DISABLE_MSVC_WARNING(4146)
```

Comment:
It seems that some time ago (JDK-8254072) 4146 was disabled for JVM MSVC build:
https://github.com/openjdk/jdk/blame/8531fa146be1da5e96c0f23091882a27c67d7893/make/hotspot/lib/CompileJvm.gmk#L117

So the warning isn't a concern after all. Sorry for the false alarm and
misleading guidance.

This was also not the correct way to introduce the warning suppression had it
been needed, but that's moot.